### DISCUSSION TITLE
Using trusted publishing and poetry

### DISCUSSION THREAD

**Discussion by tschm** [python-poetry/poetry#7859] [Show and tell]
Hoi zame,

Trusted publishing is a rather new (Apr 20) way to publish code on pypi.org. It relies on a key exchange between your package and pypi.org. This is somewhat subtle and we have implemented a Github action to simplify this problem. It works great for poetry and no longer you need to create keys on pypi and move them over as Github secrets.

A full example and a complete discussion are here:
https://github.com/marketplace/actions/pypi-token-mint

**Comment by edgarrmondragon** [comment]
The official [PyPA action](https://github.com/pypa/gh-action-pypi-publish) works fine as well.
